branch-checker-action.js

```diff
@@ -0,0 +1,103 @@
+// https://github.com/actions/toolkit/tree/main/packages/core#annotations
+const core = require('@actions/core')
+
+const { findIssueNumber, configReader, BaseAction } = require('gh-action-components')
+
+class BranchCheckerAction extends BaseAction {
+
+	constructor(options = {}) {
+		super()
+		this.options = options
+	}
+
+	async runAction() {
+		await this.checkAgainstBranchName()
+		await this.checkAgainstIssueMilestone()
+	}
+
+	/**
+	 * If the branch is one that resolves merge conflicts, extract the target
+	 * branch alias from the name isntead of using the Issue milestone.
+	 *
+	 * https://github.com/SpiderStrategies/Scoreboard/pull/44997#issuecomment-1026432096
+	 */
+	async checkAgainstBranchName() {
+		const { configFile, baseBranch, prBranch, prAuthor } = this.options
+		const regexResult = /issue-\d*-pr-\d*-conflicts-([\w|-]*)$/g.exec(prBranch)
+		const branchAlias = (regexResult && regexResult.length > 1) ? regexResult[1] : undefined
+		if (branchAlias) {
+			this.done = true
+			const { branchByAlias } = configReader(configFile)
+			const { name } = branchByAlias[branchAlias] || {}
+			if (baseBranch != name) {
+				const msg = `${prAuthor} This pull request is against the wrong branch. It must be \`${name}\` instead of \`${baseBranch}\``
```

Comment:
@mattsgarlata Made this message more assertive since we know these should always be against a given branch.

Reply:
Nice, thanks!